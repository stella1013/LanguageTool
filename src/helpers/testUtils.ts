//import { ConnectedRouter } from 'connected-react-router';
//import React from 'react';
//import { Provider } from 'react-redux';
////import { mergeDeepRight } from 'ramda';
//import rootReducer from '../redux/reducers';
////import  createStoreWithMiddleWare  from '../redux/store';
//import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
//import { AppState } from '../redux/store';
//import ReduxThunk  from 'redux-thunk';
//import { createStore, applyMiddleware } from 'redux';
//import rootReducer from '../redux/reducers';
//import thunk from 'redux-thunk';

/*
	* Returns Shallow Wrapper containing node(s) with the given data-test value
	* @param {Shallow Wrapper} - Enzyme shallow wrapper to search within
	* @param {string} val - Value of data-test attribute for search.
	* @returns {Shallow Wrapper}
	*/ 
	export const findByTestAttr = (wrapper:any, val:string) => {
		return wrapper.find(`[data-test="${val}"]`);
	}

	export const storeFactory = (initialState:any) =>{
		const createMockStore = configureMockStore();
		return createMockStore(initialState);
	}
	/*
	export const makeMountRender = (Component:any, defaultProps = {}) => {
		return (customProps = {}) => {
		  const props = {
			...defaultProps,
			...customProps
		  };
		  return mount(<Component {...props} />);
		};
	  };

	  export const makeStore = (customState = {}) => {
		const root = rootReducer({}, { type: '@@INIT' });
		const state = mergeDeepRight(root, customState);
		return createStoreWithMiddleWare(rootReducer, state);
	  };
	  export const snapshotify = (reactWrapper:any) => {
		return reactWrapper.html();
	  };

	  export const reduxify = (Component, props = {}, state = {}) => {
		return function reduxWrap() {
		  return (
			<Provider store={makeStore(state)}>
			  <ConnectedRouter history={history}>
				<Component {...props} />
			  </ConnectedRouter>
			</Provider>
		  );
		}
	  };

*/