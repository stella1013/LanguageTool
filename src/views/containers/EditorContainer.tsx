import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../redux/store';
import { Card, Row, Button } from 'react-bootstrap';
import AppTitleContainer from './AppTitleContainer';
import ProgressBarContainer from './ProgressBarContainer';
import PageListContainer from './PageListContainer';
import { modeOperations } from '../../redux/ModeController';
import { ScreenTypes } from '../../redux/ModeController/constants';

class EditorContainer extends React.Component<ReduxType> {
	backButtonHandler = (e: React.MouseEvent) => {
		this.props.changeAppMode(ScreenTypes.DASHBOARD_MODE);
	};
	render() {
		return (
			<Row className="justify-content-md-center" data-test="component-editor">
				<Card bg="light" style={{ width: '60rem' }}>
					<Card.Body>
					<Card.Header><div className="row">
							<div className="col-3-of-4"><AppTitleContainer title={this.props.title} data-test="apptitle" /></div>
							<div className="col-1-of-4"><Button variant="light"
							onClick={(e: React.MouseEvent) => this.backButtonHandler(e)}
							data-test="loadform-button"
						>
							Main Menu
						</Button></div>
						</div></Card.Header>
						
						
						
						<ProgressBarContainer />
						<PageListContainer />
					</Card.Body>
				</Card>
			</Row>
		);
	}
}
const mapStateToProps = (state: AppState) => {
	return {
		title: state.appTitle.appTitle
	};
};
const mapDispatchToProps = (dispatch: any) => {
	return {
		changeAppMode: (mode: ScreenTypes) =>
			dispatch(modeOperations.changeMode(mode))
		//showModal: (showModal:boolean) => dispatch(modalOperations.showModal(showModal))
	};
};
type ReduxType = ReturnType<typeof mapStateToProps> &
	ReturnType<typeof mapDispatchToProps>;

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(EditorContainer);
