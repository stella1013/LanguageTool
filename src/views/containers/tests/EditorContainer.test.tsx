import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import configureMockStore from 'redux-mock-store';
import { findByTestAttr } from '../../../helpers/testUtils';
import thunk from 'redux-thunk';
import EditorContainer from '../EditorContainer';
import { Provider } from 'react-redux';

describe('Editor', () => {
	/*
	 * Factory function to create a Shallow Wrapper for the App Component
	 * @function setup
	 * @param {object} props - component props
	 * @param {object} state - initial state
	 * @returns {Shallow Wrapper}
	 */
	const createMockStore = configureMockStore([thunk]);
	const initialState = {};
	//const store = createMockStore(initialState);
	let wrapper: any;
	describe('rendering', () => {
		let props ={
			appTitle:'title'
		}
		beforeEach(() => {
			const store = createMockStore(initialState);
			wrapper = shallow(<Provider store={store}><EditorContainer  /></Provider>)
			
		});

		it('renders without crashing', () => {
			const component = findByTestAttr(wrapper, 'component-editor');
			expect(component).toBeTruthy();
		});

		it('render matches snapshot', () => {
			expect(wrapper).toMatchSnapshot();
		});
		it('renders AppTitle', () => {
			const appTitle = findByTestAttr(wrapper, 'apptitle');
			expect(appTitle).toBeTruthy();
		});
		/*
		it('renders modal Toggle button', () => {
			const modalButton = findByTestAttr(wrapper, 'modaltoggle-button');
			expect(modalButton.length).toBe(1);
		});

		it('renders Modal', () => {
			const modal = findByTestAttr(wrapper, 'modalDisplay');
			expect(modal.length).toBe(1);
		});
		it('clicking display specs button will show Modal', () => {
			//find button and click
			const modalButton = findByTestAttr(wrapper, 'modaltoggle-button');
			modalButton.simulate('click');
			wrapper.update();
			//find modalstate and test value
			const initModalState = wrapper.state('showModal');
			expect(initModalState).toBe(true);
		});
		*/
	});
});
