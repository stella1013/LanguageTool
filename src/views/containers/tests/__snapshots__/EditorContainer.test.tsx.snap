// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor rendering render matches snapshot 1`] = `ShallowWrapper {}`;
